pkg/volume/util/operationexecutor/operation_executor_test.go

```diff
@@ -224,3 +354,49 @@ func getTestPodWithGCEPD(podName, pdName string) *v1.Pod {
 		},
 	}
 }
+
+func isOperationRunSerially(ch, quit chan interface{}) bool {
+	defer close(quit)
+	numOperationsStarted := 0
+loop:
+	for {
+		select {
+		case <-ch:
+			numOperationsStarted++
+			if numOperationsStarted > 1 {
+				return false
+			}
+		case <-time.After(5 * time.Second):
+			break loop
+		}
+	}
+	return true
+}
+
+func isOperationRunConcurrently(ch, quit chan interface{}, numOperationsToRun int) bool {
+	defer close(quit)
+	numOperationsStarted := 0
+loop:
+	for {
+		select {
+		case <-ch:
+			numOperationsStarted++
+			if numOperationsStarted == numOperationsToRun {
+				return true
+			}
+		case <-time.After(5 * time.Second):
+			break loop
+		}
+	}
+	return false
+}
+
+func setup() (chan interface{}, chan interface{}, OperationExecutor) {
+	ch, quit := make(chan interface{}), make(chan interface{})
+	return ch, quit, NewOperationExecutor(newMockOperationGenerator(ch, quit))
```

Comment:
If you want to do some more clean up, move the "mockOperationGenerator" in to its own file. Also is it a mock or a fake? I forget what the difference is.